istioctl/pkg/validate/validate_test.go

```diff
@@ -104,15 +104,20 @@ spec:
   - handler: handler-for-valid-rule.denier
     instances:
     - instance-for-valid-rule.checknothing`
-	invalidUnsupportedKey = `
```

Comment:
Not sure why we're removing this test case? Is it subsumed?

Reply:
This test is not valuable since validation of k8s envelope is better left to k8s. The tool should be focused on istio config validation, not duplicating the work of other tools.

Reply:
It is also incorrect to assume that "spec" and "status" are the only fields in the object. That is not part of k8s specification, see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#endpoints-v1-core, for example.

Reply:
We should keep the check for unknown top-level keys, e.g. `unexpected_junk:`.  The reason is that a common user error is incorrect YAML indentation on optional fields.  Users consistently un-indent fields to the level of `Spec` and then wonder why their field is not respected.

Reply:
Restored the extra key test for pilot and mixer configs. PTAL.